### ISSUE TITLE
FreeTDS Pool segmentation fault with php client connection 

### ISSUE THREAD

**Issue by misterion** [FreeTDS/freetds#33]
FreeTDS Pool segmentation fault on connection to it with [mssql](http://php.net/manual/en/book.mssql.php) or [pdo](http://php.net/manual/ru/ref.pdo-dblib.php) api in php. 

Using the following code pdo code

     $host = "your pool ip";
     $port = "your pool port";
     $login = "...";
     $password = "...";
     $link = new PDO("dblib:host=$host:$port", $login, $password);

You got error `PHP Fatal error:  Uncaught exception 'PDOException' with message 'SQLSTATE[01002] Adaptive Server connection failed (severity 9)'`.

Using the following mssql api code

   $link = mssql_connect("$host:$port ", $login, $password);

You got warning `PHP Warning:  mssql_connect(): Unable to connect to server`.

Both connection try cause freetds pool segmentation fault if login and password not empty. If using empty login/password freetds pool do not  segmentation fault  but php client has same errors as above writen.
     
The following link https://www.dropbox.com/s/trcx5qvmfpi4bpc/freetds.zip to freetds pool application, config and core dump files. Used freetds from master. Same with 0.95/0.99 branches.

**Comment by freddy77** [comment]
I think the problem is database setting which is missing. Could you try adding this setting. I'll come with a patch.

**Comment by misterion** [comment]
Yes, i could if you write which settings is missing.

**Comment by freddy77** [comment]
There is a "database" pool entry setting which should be the name of the database inside the database server you want

**Comment by misterion** [comment]
[mssql](http://php.net/manual/en/book.mssql.php) api allow to select data base only after connection established so i can`r test it with your hypothesis. PDO connection produce same crash with/without database name in dsn string.

**Comment by freddy77** [comment]
Sorry, there is a setting in pool.conf, under the section of file referring to your pool. You should add a "database = XXX" line.

**Comment by misterion** [comment]
Yes, this fix crash. But connectiom from these php api don`t work. The error is "SQLSTATE[HY000] Bad token from the server: Datastream processing out of sync (severity 9)". This look`s like more one new issue ?

**Comment by misterion** [comment]
The freetds.conf settings: 
```ini
[global]
[global]
 tds version = 7.2
 client charset = UTF-8
 dump file = /var/log/freetds.log
 debug flags = 0xffff
 text size = 4194304

[vmdb]
host = vm-db 
port = 1433 
tds version = 7.2 
client charset = UTF-8
```

Here is TDSDUMP with php pdo and mssql connections - https://www.dropbox.com/s/vaxjc81r9kd3mwb/tdsdump.zip

**Comment by freddy77** [comment]
Sorry, I saw your reply just noew.

Ok, from the logs looks like the PHP is connecting with 7.1 protocol while pool uses 7.2 so you get a mismatch and server/PHP close automatically the connection.

Now, this check is not done by pool (still in my todo list), on the other way from your configuration is not clear why PHP is using 7.1 instead of 7.2. Usually the problem with PHP is that it does not read the freetds.conf you are expecting because is executed with a different user.

Looks like all dumps you provided are from the pool which tried to reset the connection (only the pool code use this flag).

**Comment by misterion** [comment]
I`l check it and think PHP side is not a problem. I test with one freetds.conf file, launch pool and php under one user in cli. If using 7.1 protocal - all work fine in pool or direct connection with freetds. If using 7.2, 7.3, 7.4 all works well in direct connection, and fail with pool with same problem as i post above. 

Today i update freetds from master and got new issue with this case. While using 7.1 - all works, if using >7.1 pool fail with assert:
```
Found conf file in /usr/local/etc/pool.conf reading sections
Listening on port 5000
accepting connection
error: login.c:219: tds7_read_login: tds_iconv() failed
tdspool: login.c:220: tds7_read_login: Assertion `-1 != a' failed.
Aborted
```
p.s. here is logs https://www.dropbox.com/s/qqu14r0wi8ta057/freetds_log.zip

**Comment by freddy77** [comment]
Working on it. The login implementation in the server part did not support correctly all versions.

**Comment by misterion** [comment]
We working on it too. Wait for our pull request about this issue in 3-4 days.

**Comment by freddy77** [comment]
You will have some conflict probably. Don't forget to rebase on last master.

**Comment by freddy77** [comment]
Fixed in master.
